store/src/state.rs

```diff
@@ -355,10 +356,9 @@ impl State {
                 .chain_mmr
                 .get_delta(block_num as usize, state_sync.block_header.block_num as usize)?;
 
-            let proof = inner.chain_mmr.open(
-                state_sync.block_header.block_num as usize,
-                state_sync.block_header.block_num as usize + 1,
-            )?;
+            let proof = inner
+                .chain_mmr
+                .open(block_num as usize, state_sync.block_header.block_num as usize)?;
```

Comment:
The merkle path shouldn't be necessary. The client can just add the last block as a new peak, and track it. All the information needed is available:

- on a new update: the left siblings of the merkle path are the existing mmr peaks, the right siblings are from the new delta.
- on the add mentioned above: explained here https://github.com/0xPolygonMiden/miden-node/issues/152#issuecomment-1894597087